system/jlib/jsecrets.cpp

```diff
@@ -1119,19 +1119,30 @@ static IPropertyTree * resolveLocalSecret(const char *category, const char * nam
     {
         if (entries->isDir())
             continue;
-        StringBuffer name;
-        entries->getName(name);
-        if (!validateXMLTag(name))
+        StringBuffer secretKeyName;   // filename is used as the secret key name
+        entries->getName(secretKeyName);
+        if (!validateXMLTag(secretKeyName))
             continue;
-        MemoryBuffer content;
+
         Owned<IFileIO> io = entries->query().open(IFOread);
-        read(io, 0, (size32_t)-1, content);
-        if (!content.length())
+        if (!io)
+            continue;
+
+        offset_t sz = io->size();
+        if (sz == 0)
             continue;
 
-        //Always add a null terminator to data read from a file so that queryProp() can be used on the resultant tree
-        content.append((byte)0);
-        tree->setPropBin(name, content.length(), content.bufferBase());
+        StringBuffer value;
+        char *valuePtr = value.reserveTruncate(sz);
+        if (sz != io->read(0, sz, valuePtr))
+            continue;
+
+        *(valuePtr+sz) = '\0';
```

Comment:
@kenrowland This code shouldn't be needed, and wasn't in the code that jake approved.  Was it added to fix a problem?

Reply:
Yes. It properly terminated the string so that the clip worked.

Reply:
I am going to verify, but it may have been to ensure proper termination. The previous code suggestion did not properly terminate the string that was read and clip did not work.

Reply:
The line was to ensure the string was terminated, but not needed when using the StringBuffer. Line removed.